Sources/Logging/Logging.swift

```diff
@@ -505,30 +505,72 @@ public struct MultiplexLogHandler: LogHandler {
     }
 }
 
-/// Ships with the logging module, really boring just prints something using the `print` function
-internal struct StdoutLogHandler: LogHandler {
-    private let lock = Lock()
+/// A wrapper to facilitate `print`-ing to stderr and stdio that
+/// ensures access to the underlying `FILE` is locked to prevent
+/// cross-thread interleaving of output.
+internal struct StdioOutputStream: TextOutputStream {
+    internal let file: UnsafeMutablePointer<FILE>
+
+    internal func write(_ string: String) {
+        string.withCString { ptr in
+            flockfile(file)
+            defer {
+                funlockfile(file)
+            }
+            _ = fputs(ptr, file)
+        }
+    }
+
+    internal static let stderr = StdioOutputStream(file: systemStderr)
+    internal static let stdout = StdioOutputStream(file: systemStdout)
+}
 
-    public init(label: String) {}
+// Prevent name clashes
+#if os(macOS) || os(tvOS) || os(iOS) || os(watchOS)
+let systemStderr = Darwin.stderr
+let systemStdout = Darwin.stdout
+#else
+let systemStderr = Glibc.stderr!
+let systemStdout = Glibc.stdout!
+#endif
 
-    private var _logLevel: Logger.Level = .info
+/// `StreamLogHandler` is a simple implementation of `LogHandler` for directing
+/// `Logger` output to either `stderr` or `stdout` via the factory methods.
+public struct StreamLogHandler: LogHandler {
 
-    public var logLevel: Logger.Level {
+    /// Factory that makes a `StreamLogHandler` to directs its output to `stdout`
+    public static func makeStdoutLogHandler(label: String) -> StreamLogHandler {
```

Comment:
naming nit: i like the new factories but dont love the names. maybe `makeStdout` and `makeStderr` is enough. in any case not a big deal, so can go with this too

Reply:
You're right, it is a bit repetitive in the full context `StreamLogHandler.makeStdoutLogHandler`, happy to change. While you're considering names, I made up the `StreamLogHandler` name to convey it's new expanded capabilities, but it could just as well be `TextOutputStreamLogHandler` :-)

Reply:
naming is hard. @weissi any preferences?

Reply:
@tomerd not really but I agree with both of you that `makeStdoutLogHandler` is a bit long and redundant. We could even do something like `standardOutput` only. Then it would read

    Logging.bootstrap(StreamLogHandler.standardOutput)

`makeStdout` sounds like it's making a 'stdout' to me...